docfiles/apptracking.html

```diff
@@ -29,14 +30,29 @@
             }
         });
 
-        var scriptIdRegex = /(?:S?\d{5}-\d{5}-\d{5}-\d{5})|(?:_[0-9a-zA-Z]{12})/g;
         // Check if the current page contains a share URL
         function isScriptPage(url) {
-            return !!url.match(scriptIdRegex);
+            return !!url.match(window.scriptIdRegex);
         }
-        // Scrub the key (if any) from the URL.
+
+        // Scrub potential PII from the URL.
         function scrubUrl(url) {
-            return url.replace(scriptIdRegex, "xxxxx-xxxxx-xxxxx-xxxxx");
+            debugger;
+            if(!url) return url;
+
+            var scrubbedUrl = new URL(url.toLocaleLowerCase().replace(window.scriptIdRegex, "xxxxx-xxxxx-xxxxx-xxxxx"));
+
+            // Just as a precaution, strip any suspicious query parameters.
+            let blockedParamKeywords = ["username", "password", "token", "pwd"];
```

Comment:
are these coming from somewhere? I'd be more concerned about the parameters themselves in the first place as the query parameters are part that go to the server / would probably show up in backend logs? I'm not sure what scrubbing them would do for us really, as it would be just like going to `makecode.com/{my-credit-card-number}` or the user explicitly calling window.pxtTickEvent from the dev tools

Reply:
This function gets used to clean the urlReferrer property (among other things), which could be any website with any params that ultimately linked to our site. We don't really know or control it in that case. And frankly I'd hope they aren't including PII in the url anyway...but this is meant to be an extra precaution just in case (and it might help reduce false positives on automated checks, like one we received the other day from something that looked sort of like a password but wasn't).

Reply:
can we just cut off all the query parameters from the referrer then? I doubt we care about anything more than the domain and maybe path? Any sequence of unicode characters with length > ~6 'looks' like a possible password to some extent

Reply:
I considered that, but apparently some of them are important.

Reply:
@abchatra might be able to provide more context on which ones

Reply:
Minecraft has bunch of query parameters like Minecraft version and protocol version, tutorial md etc. it is very useful to query those

Reply:
Maybe allowlist known good ones domains vs trying to denylist arbitrary input? It sounds like the issue came from referrer which is a bit different from our url (I was suggesting more aggressive scrubbing of urlReferrer than url, as url is 'explicitly intended for us', and minecraft params are all queryParams on url)

I'll defer to you though @abchatra , probably not something that really comes up too much in reality

Reply:
I agree this may not be very useful. However, I am ok with this change as it would atleast avoid these false positives.